src/domain/new_address_data.rs

```diff
@@ -0,0 +1,23 @@
+
+#[derive(serde::Deserialize)]
+pub struct AddressData {
+    pub user_id: i32,
+    pub derivation_path: String,
+    pub child_pubk_1: String,
+    pub child_pubk_2: String,
+    pub service_pubk: String,
+}
+
+#[derive(sqlx::Type)]
+pub struct NewAddressData {
+    pub user_id: i32,
+    pub derivation_path: String,
```

Comment:
Any String type will be valid here and in the other fields, i.e. child_pubk_1, child_pubk_2, and service_pubk. We should always stick with type-driven development. There's an Xpub type that the extended public keys can be converted into. The same should apply for derivation_path. In that way we know we are dealing with valid types that were parsed from String.